### ISSUE TITLE
elements that allow any aria role flag 4 aria-dpub roles as not allowed

### ISSUE THREAD

**Issue by scottaohara** [dequelabs/axe-core#3176]
Product: axe-core 

Expectation: Missing support for dpub aria roles `doc-backlink`, `doc-biblioref`, `doc-glossref`, and `doc-noteref`

Actual: on elements that allow any ARIA role, the above 4 are consistently flagged as not being allowed for the given element.  Would venture to guess these four were just missed in the inclusion of the other dpub roles which do not get flagged.

A [sample page of the `b` element](https://thepaciellogroup.github.io/aria-test-builder/dist/testcases/b-test.html) - running axe flags the mentioned roles.

<details>
<summary>These elements allow any role, and would need to support these 4 roles:</summary>
<ul><li>a no href
<li>abbr
<li>bbdi
<li>bdo
<li>blockquote
<li>canvas
<li>cite
<li>code
<li>data
<li>del
<li>dfn
<li>div
<li>em
<li>figure without descendant figcaption
<li>hgroup
<li>i
<li>ins
<li>kbd
<li>mark
<li>output
<li>p
<li>pre
<li>q
<li>small
<li>span
<li>strong
<li>svg
<li>sub
<li>sup
<li>u
<li>var
</ul>
</details>

Motivation: there should be no exclusion of allowed roles when an element allows any role.

<pre><code>axe-core version: 4.3.3
</code></pre>

**Comment by WilcoFiers** [comment]
Gotcha, okay yes this is intentional. Axe-core is failing these because these roles aren't widely supported. Specifically, what axe-core is doing is to fail DPUB roles that inherit from link, if they are used on elements that don't have an implicit role of `link`. This has to do with accessibility support.

```html
<a href="#" role="doc-backlink">back</a> <!-- pass -->
<b role="doc-backlink" tabindex="0">back</b> <!-- fail -->
```

We are looking to change how accessibility support works in axe-core, so we may very well end up changing how these kinds of cases get reported.

**Comment by scottaohara** [comment]
Thanks for the background @WilcoFiers.

**Comment by scottaohara** [comment]
understood.  thanks again for the background.
